examples/decoupled/repeat_client.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import sys
+from functools import partial
+import numpy as np
+
+from tritonclient.utils import *
+import tritonclient.grpc as grpcclient
+
+
+class UserData:
+
+    def __init__(self):
+        self._completed_requests = queue.Queue()
+
+
+def callback(user_data, result, error):
+    if error:
+        user_data._completed_requests.put(error)
+    else:
+        user_data._completed_requests.put(result)
+
+
+# This client sends a single request to the model with the
+# following tensor data. In compliance with the behavior
+# of repeat_int32 model, it will expect the 4 responses
+# with output: [4], [2], [0] and [1] respectively.
+model_name = "repeat_int32"
+in_value = [4, 2, 0, 1]
+delay_value = 2
+wait_value = 5
+
+inputs = []
+inputs.append(grpcclient.InferInput('IN', [len(in_value)], "INT32"))
+inputs.append(grpcclient.InferInput('DELAY', [1], "UINT32"))
+inputs.append(grpcclient.InferInput('WAIT', [1], "UINT32"))
+
+outputs = []
+outputs.append(grpcclient.InferRequestedOutput('OUT'))
+outputs.append(grpcclient.InferRequestedOutput('IDX'))
+
+with grpcclient.InferenceServerClient(url="localhost:8001",
```

Comment:
Okay syntax for Python, but should wrap the process in `if __name__ == '__main__':`

Reply:
I didn't wrap the script in main because none of our client scripts in the example directory are currently within main.
@Tabrizian Was this intentional? If yes, then why?

Reply:
There is not any particular reason. We can wrap those clients in `if __name__ == '__main__'` too.

Reply:
Should use `__main__` then, otherwise the code will be executed if the code is accidentally imported in other Python script.

Reply:
I will do this in another PR.